JAVA-1200: Upgrade LZ4 to 1.3.0.

adutra opened the pull request:
This PR also introduces a new OSGi test to validate that LZ4 bundle can be used in an OSGi environment.

In order to make this possible, I had to cherry-pick 3 commits from the 3.0.x branch:
1. 5ea6407 JAVA-1133: Add OSGi headers to cassandra-driver-extras.
2. 4d8f9b3 JAVA-1069: Bootstrap driver-examples module.
3. 297bcbc JAVA-1150: Add example and FAQ entry about ByteBuffer/BLOB.

Once these commits make into 3.0 branch then they can be safely deleted from this PR.

Commit:
JAVA-1200: Upgrade LZ4 to 1.3.0.

This commit also upgrades Snappy version to 1.1.2.6 and Hdr Histogram version to 2.1.9.

Besides, it also introduces a new OSGi test to validate that LZ4 bundle can be used in an OSGi environment.
As this new test relies on a system property that might be present or not, this commit switches the Pax Exam container from "native" to "forked", because native containers are not confined enough to isolate system properties correctly.

olim7t commented:
LGTM.
We should also upgrade Snappy while we're at it. We're using 1.0.5, the latest is 1.1.2.4.

Commit:
JAVA-1209: ProtocolOptions.getProtocolVersion() should return null
instead of throwing NPE if Cluster has not been init'd

adutra committed:
JAVA-1147: Upgrade Netty to 4.0.37. (#694)

This commit also solves JAVA-1213: Allow updates and inserts to BLOB
column using read-only ByteBuffer.

olim7t committed:
Merge branch 'dmitrymurashenkov-java1209' into 3.0.x

tolbertam committed:
JAVA-1204: Update documentation to indicate tcnative version requirement. (#687)

adutra commented:
Rebased on top of 3.0.x and had to close this in favor of #698, sorry.
